From df76963a5cd56c677f189ad89d75935a76fca090 Mon Sep 17 00:00:00 2001
From: Robin Ole Heinemann <robin.ole.heinemann@t-online.de>
Date: Sun, 13 Dec 2020 02:22:57 +0100
Subject: [PATCH] anv: Add DRM_RDWR flag in anv_gem_handle_to_fd

The DRM_RDWR flag is needed for mmap with PROT_WRITE to work.

Cc: mesa-stable
Signed-off-by: Robin Ole Heinemann <robin.ole.heinemann@gmail.com>
Reviewed-by: Lionel Landwerlin <lionel.g.landwerlin@intel.com>
Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/8075>
---
 src/intel/vulkan/anv_gem.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/intel/vulkan/anv_gem.c b/src/intel/vulkan/anv_gem.c
index 81e22428e18..4c43f5f77b4 100644
--- a/src/intel/vulkan/anv_gem.c
+++ b/src/intel/vulkan/anv_gem.c
@@ -437,7 +437,7 @@ anv_gem_handle_to_fd(struct anv_device *device, uint32_t gem_handle)
 {
    struct drm_prime_handle args = {
       .handle = gem_handle,
-      .flags = DRM_CLOEXEC,
+      .flags = DRM_CLOEXEC | DRM_RDWR,
    };
 
    int ret = gen_ioctl(device->fd, DRM_IOCTL_PRIME_HANDLE_TO_FD, &args);
-- 
2.29.2

